### PR TITLE
Working selector as item in row

### PR DESCRIPTION
**Pull request by tgsoverly**
Wrapped the `CohortSelect` model in a new `Selector` component to display them inline.

A bigger change in this is to allow "options" to be trace references.  This seems like an easy quick step toward having the more complex "meta" tagging we talked about.

### CODE REVIEW

File: viewer/src/components/Dashboard.js [hunk 1]
```diff
@@ -60,6 +62,13 @@ const Dashboard = (props) => {
                 width={getWidth(row, item)}
                 height={getHeight(row.height)}
                 key={`dashboardRow${rowIndex}Item${itemIndex}`} />
+        } else if (item.selector) {
```
Comment (tgsoverly) [inline viewer/src/components/Dashboard.js:65]: Add dashboard test render for selector

File: viewer/src/components/items/Selector.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import React from "react";
+import CohortSelect from "./CohortSelect";
+import tw from "tailwind-styled-components"
+import { useTracesData } from "../../hooks/useTracesData";
+
+export const ChartContainer = tw.div`
+    relative
+`;
+
+const Selector = ({ selector, project, itemWidth }) => {
```
Comment (tgsoverly) [inline viewer/src/components/items/Selector.js:11]: Add test

File: visivo/models/item.py [hunk 3]
```diff
@@ -44,19 +47,23 @@ def id(self):
...
     )
+    selector: Optional[generate_ref_field(Selector)] = Field(
+        None, description="A Selector object defined inline or a ref() to a selector"
+    )
 
     @model_validator(mode="before")
     @classmethod
     def validate_column_refs(cls, data: any):
-        markdown, chart, table = (
+        markdown, chart, table, selector = (
```
Comment (tgsoverly) [inline visivo/models/item.py:57]: Add test

File: visivo/models/selector.py [hunk 4]
```diff
@@ -31,17 +33,30 @@ class Selector(NamedModel, BaseModel):
...
     """
-    model_config = ConfigDict(extra='ignore')  
+
+    model_config = ConfigDict(extra="ignore")
 
     name: str = Field(description="The name of the selector")
     type: SelectorType = Field(
         SelectorType.multiple, description="Single or multiple selector"
     )
+    options: List[RefString] = Field(
```
Comment (tgsoverly) [inline visivo/models/selector.py:43]: Should this have validation?

File: visivo/parsers/serializer.py [hunk 5]
```diff
@@ -52,6 +52,10 @@ def replace_item_ref(item):
                             trace.model.models = ParentModel.all_descendants_of_type(
                                 type=Model, dag=dag, from_node=trace.model
                             )
+                if item.selector:
```
Comment (tgsoverly) [inline visivo/parsers/serializer.py:55]: Test

File: visivo/models/selector.py [hunk 6]
```diff
@@ -31,17 +33,30 @@ class Selector(NamedModel, BaseModel):
...
     def serialize_model(self):
-        return {"name": self.name, "type": self.type, "parent_name": self._parent_name}
+        model = {"name": self.name, "type": self.type, "options": self.options}
+        if hasattr(self, "_parent_name"):
+            model["parent_name"] = self._parent_name
+        else:
+            model["parent_name"] = self.name
+        return model
+
+    def child_items(self):
```
Comment (tgsoverly) [inline visivo/models/selector.py:61]: Add dag test

File: visivo/models/item.py [hunk 7]
```diff
@@ -22,11 +23,14 @@ class Item(BaseModel, ParentModel):
         table: ref(table-name)
       - width: 2
         chart: ref(chart-name)
+      - width: 1
+        chart: ref(selector-name)
```
Comment (jpjsonic) [inline visivo/models/item.py:27]: Did you mean to say selector here rather than chart for the key?

File: visivo/models/row.py [hunk 8]
```diff
@@ -7,6 +7,7 @@
 
 
 class HeightEnum(str, Enum):
```
Comment (jpjsonic) [inline visivo/models/row.py:9]: I wonder if we should also allow people to input pixels too if they want to be more specific about the height.

**Review by jpjsonic — APPROVED** [review]
Looks great!

File: visivo/models/item.py (hunk 7 above)
Comment (tgsoverly) [reply visivo/models/item.py:27]: I did!

File: visivo/models/row.py (hunk 8 above)
Comment (tgsoverly) [reply visivo/models/row.py:9]: I am having some layout issues, so I think I need to come back and fix them in a new PR.

**Pull request merged**